app/models/green_lanes/measure.rb

```diff
@@ -7,5 +7,8 @@ class Measure < Sequel::Model(:green_lanes_measures)
     many_to_one :goods_nomenclature, class: 'GoodsNomenclature',
                                      primary_key: %i[goods_nomenclature_item_id producline_suffix],
                                      key: %i[goods_nomenclature_item_id productline_suffix]
+    def goods_nomenclature_id
+      goods_nomenclature_item_id
+    end
```

Comment:
I cannot see this used anywhere so can this be removed I think?

Reply:
This is checked by - app/serializers/api/admin/green_lanes/measure_serializer.rb

`has_one :goods_nomenclature, serializer: GoodsNomenclatureSerializer`

I think that's the convention to map this association as the name is goods_nomenclature. If the name is goods_nomenclature_item, then this wouldn't have been an issue.

Reply:
Ah, now I understand - I think you'll want to use a uniq id for the relationship, item id isn't uniq at a given point in time but sid is

`has_one :goods_nomenclature, serializer: GoodsNomenclatureSerializer, id_method_name: :goods_nomenclature_sid`

You'll also want to set the id on GoodsNomenclatureSerializer to be `goods_nomenclature_sid` instead of `goods_nomenclature_item_id`

Reply:
https://github.com/jsonapi-serializer/jsonapi-serializer?tab=readme-ov-file#customizable-options

Reply:
Thanks, changed the id to `goods_nomenclature_sid `